viewer/src/main/java/nl/b3p/viewer/stripes/FeatureInfoActionBean.java

```diff
@@ -354,7 +359,210 @@ public Resolution info() throws JSONException {
+                        ClobElement ce = layer.getService().getDetails().get(GeoService.DETAIL_USE_INTERSECT);
+                        useIntersect = Boolean.parseBoolean(ce.getValue());
+                    }
+                    if (!useIntersect) {
+                        Point point = new GeometryFactory().createPoint(new Coordinate(
+                                Double.parseDouble(x),
+                                Double.parseDouble(y)));
+
+                        spatialFilter = ff.dwithin(ff.property(geomAttribute), ff.literal(point), Double.parseDouble(distance), "meters");
+                    } else {
+                        GeometricShapeFactory shapeFact = new GeometricShapeFactory();
+                        shapeFact.setNumPoints(32);
+                        shapeFact.setCentre(new Coordinate(
+                                Double.parseDouble(x), Double.parseDouble(y)));
+                        shapeFact.setSize(Double.parseDouble(distance) * 2);
+                        Polygon p = shapeFact.createCircle();
+                        spatialFilter = ff.intersects(ff.property(geomAttribute), ff.literal(p));
+                    }
+
+                    Filter currentFilter = filter != null && filter.trim().length() > 0 ? CQL.toFilter(filter) : null;
+
+                    if (currentFilter != null) {
+                        currentFilter = (Filter) currentFilter.accept(new ChangeMatchCase(false), null);
+                    }
+
+                    Filter f = currentFilter != null ? ff.and(spatialFilter, currentFilter) : spatialFilter;
+                    //only remove unit if it is a JDBC datastore
+                    if (JDBCFeatureSource.PROTOCOL.equals(layer.getService().getProtocol())) {
+                        f = (Filter) f.accept(new RemoveDistanceUnit(), null);
+                    }
+
+                    f = FeatureToJson.reformatFilter(f, layer.getFeatureType());
+
+                    q.setFilter(f);
+                    q.setMaxFeatures(limit + 1);
+
+                    JSONArray features = executeQuery(al, layer.getFeatureType(), fs, q);
+                    if (features.length() > limit) {
+                        JSONArray newArray = new JSONArray();
+                        for (int j = 0; j < features.length(); j++) {
+                            if (j < limit) {
+                                newArray.put(features.get(j));
+                            }
+                        }
+                        features = newArray;
+                        response.put("moreFeaturesAvailable", true);
+                    }
+                    jFeat = features.getJSONObject(0);
+                    response.put("features", features);
+                } while (false);
+            } catch (Exception e) {
+                log.error("Exception loading feature info for " + exceptionMsg, e);
+                error = "Exception: " + e.toString();
```

Comment:
since the message is returned to the front end an may be displayed to a user it is probably better to use `e.getLocalizedMessage()` and not `toString()`
--


`